pybaseball/utils.py

```diff
@@ -281,10 +290,23 @@ def flag_imputed_data(statcast_df: pd.DataFrame) -> pd.DataFrame:
     return df_return
 
 def norm_pitch_code(pitch: str, to_word: bool = False) -> str:
-    normed = name_to_code_map.get(pitch.upper())
-    normed = code_to_name_map.get(normed) if to_word else normed
+    normed = pitch_name_to_code_map.get(pitch.upper())
+    normed = pitch_code_to_name_map.get(normed) if to_word else normed
     if normed is None:
         if pitch.lower() == 'all':
             raise ValueError("'All' is not a valid pitch in this particular context!")
         raise ValueError(f'{pitch} is not a valid pitch!')
     return normed
+
+def norm_positions(posn: str, to_word: bool = False, to_number: bool = True) -> str:
+    normed = posn_name_to_code_map.get(posn.upper())
+    normed = posn_code_to_name_map.get(normed) if to_word else normed
+    if to_number:
+        if normed not in ["IF", "OF"]:
+            normed = posn_code_to_numbers_map.get(normed)
+        if posn.lower() == "all":
+            normed = ""
+    if normed is None:
+        raise ValueError(f'{posn} is not a valid position!')
+    return normed.lower()
```

Comment:
This whole thing has type problems (and IF / OF don't have numeric norms here?)

Reply:
Yeah this was tough to figure out. The Statcast url uses numbered positions for when you pick a single one, but then for `IF / OF`, the url requires lowercase `if / of` and for `all`, the url just uses an empty string. Not the best for what we're trying to do

Reply:
Yeah but you're potentially calling lower on an int

Reply:
Technically it isn't since I'm casting the ints to strings when created the map earlier in the code, so actually I didn't need to update the return types like I did - it'll always return a string. I can add some in-line comments to make that clearer or move the `lower()` call to somewhere that makes more sense